polkadot/node/core/prospective-parachains/src/fragment_chain/tests.rs

```diff
@@ -1415,37 +1709,49 @@ fn pending_availability_in_scope() {
 		vec![relay_parent_b_info],
 	)
 	.unwrap();
-	let tree = FragmentTree::populate(scope, &storage);
+	let chain = FragmentChain::populate(scope, &storage);
 
-	let candidates: Vec<_> = tree.candidates().collect();
-	assert_eq!(candidates.len(), 2);
-	assert_eq!(tree.nodes.len(), 2);
+	assert_eq!(chain.chain().len(), 2);
 
 	let candidate_d_hash = CandidateHash(Hash::repeat_byte(0xAA));
 
-	assert_eq!(
-		tree.hypothetical_depths(
-			candidate_d_hash,
-			HypotheticalCandidate::Incomplete {
-				parent_head_data_hash: HeadData::from(vec![0x0b]).hash(),
-				relay_parent: relay_parent_c,
-			},
-			&storage,
-			false,
-		),
-		vec![1],
-	);
+	assert!(chain.hypothetical_membership(
+		HypotheticalCandidate::Incomplete {
+			parent_head_data_hash: HeadData::from(vec![0x0a]).hash(),
+			candidate_relay_parent: relay_parent_a,
+			candidate_hash: candidate_a_hash,
+			candidate_para: para_id
+		},
+		&storage,
+	));
 
-	assert_eq!(
-		tree.hypothetical_depths(
-			candidate_d_hash,
-			HypotheticalCandidate::Incomplete {
-				parent_head_data_hash: HeadData::from(vec![0x0c]).hash(),
-				relay_parent: relay_parent_b,
-			},
-			&storage,
-			false,
-		),
-		vec![2],
-	);
+	assert!(!chain.hypothetical_membership(
+		HypotheticalCandidate::Incomplete {
+			parent_head_data_hash: HeadData::from(vec![0x0a]).hash(),
+			candidate_relay_parent: relay_parent_c,
+			candidate_para: para_id,
+			candidate_hash: candidate_d_hash,
+		},
+		&storage,
+	));
+
+	assert!(!chain.hypothetical_membership(
+		HypotheticalCandidate::Incomplete {
+			parent_head_data_hash: HeadData::from(vec![0x0b]).hash(),
+			candidate_relay_parent: relay_parent_c,
+			candidate_para: para_id,
+			candidate_hash: candidate_d_hash,
+		},
+		&storage,
+	));
+
+	assert!(chain.hypothetical_membership(
+		HypotheticalCandidate::Incomplete {
+			parent_head_data_hash: HeadData::from(vec![0x0c]).hash(),
+			candidate_relay_parent: relay_parent_b,
+			candidate_para: para_id,
+			candidate_hash: candidate_d_hash,
+		},
+		&storage,
+	));
```

Comment:
Not sure if it is covered by other tests, can we also add a test that checks work ok with forks on the relay chain and parachain cycles gets rejected.